Content.Server/AirlockPainter/AirlockPainterSystem.cs

```diff
@@ -0,0 +1,138 @@
+using Content.Server.DoAfter;
+using Content.Server.Popups;
+using Content.Server.UserInterface;
+using Content.Shared.AirlockPainter;
+using Content.Shared.AirlockPainter.Prototypes;
+using Content.Shared.Doors.Components;
+using Content.Shared.Interaction;
+using JetBrains.Annotations;
+using Robust.Server.GameObjects;
+using Robust.Shared.Audio;
+using Robust.Shared.Player;
+
+namespace Content.Server.AirlockPainter
+{
+    /// <summary>
+    /// A system for painting airlocks using airlock painter
+    /// </summary>
+    [UsedImplicitly]
+    public sealed class AirlockPainterSystem : SharedAirlockPainterSystem
+    {
+        [Dependency] private readonly UserInterfaceSystem _userInterfaceSystem = default!;
+        [Dependency] private readonly DoAfterSystem _doAfterSystem = default!;
+        [Dependency] private readonly PopupSystem _popupSystem = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<AirlockPainterComponent, AfterInteractEvent>(AfterInteractOn);
+            SubscribeLocalEvent<AirlockPainterComponent, ActivateInWorldEvent>(OnActivate);
+            SubscribeLocalEvent<AirlockPainterComponent, AirlockPainterSpritePickedMessage>(OnSpritePicked);
+            SubscribeLocalEvent<AirlockPainterDoAfterComplete>(OnDoAfterComplete);
+            SubscribeLocalEvent<AirlockPainterDoAfterCancelled>(OnDoAfterCancelled);
+        }
+
+        private void OnDoAfterComplete(AirlockPainterDoAfterComplete ev)
+        {
+            ev.Component.IsSpraying = false;
+            if (TryComp<AppearanceComponent>(ev.Target, out var appearance) &&
+                TryComp<PaintableAirlockComponent>(ev.Target, out PaintableAirlockComponent? airlock))
+            {
+                SoundSystem.Play(Filter.Pvs(ev.User), ev.Component.SpraySound.GetSound(), ev.User);
+                appearance.SetData(DoorVisuals.BaseRSI, ev.Sprite);
```

Comment:
Relatively minor, but you should replace `Filter.Pvs(ev.user)` with `Filter.Pvs(ev.user, entityManager: EntityManager)`.

Otherwise the filter has to unnecessarily resolve IEntityManager.